-  changed the function to (asynchronous) because we used (await) and the reason for using await is because we want first this line of code should be completely complied and run then the line ofter that.

---
- used try and catch to find the exact place of error.